core/src/actions/navigation.ts

```diff
@@ -102,16 +91,45 @@ export class NavigationList extends OptionallyAuthenticatedAction {
         value: clusterNameSetting?.value || "",
       },
       teamMember: teamMember ? await teamMember.apiData() : undefined,
-      navigationModel: isAuthenticated
-        ? {
-            value: currentModelId,
-            options: await Promise.all(models.map((m) => m.apiData())),
-          }
-        : { value: null, options: [] },
     };
   }
 
-  async authenticatedNav(teamMember: TeamMember, modelId?: string) {
+  private static createModelNavigationItems(models: GrouparooModel[]) {
+    const navigationItems: NavigationItem[] = [];
+    const hasModels = !!models.length;
+
+    if (hasModels) {
+      navigationItems.push({ type: "divider" });
+
+      models.forEach((model) => {
+        navigationItems.push({
+          type: "link",
+          title: model.name,
+          icon: model.getIcon(),
+          href: `/model/${model.id}/overview`,
+          mainPathSectionIdx: 2,
+        });
+      });
+    }
+
+    navigationItems.push({
+      type: "link",
+      title: "New Model",
+      icon: "plus",
+      href: `/model/new`,
+      mainPathSectionIdx: 2,
+      small: hasModels,
+      hideOn: ["community"],
```

Comment:
Note that you should have access to the ui edition here through `process.env.GROUPAROO_UI_EDITION`, so passing this along might not be necessary- not a huge deal though